Sources/SwiftSyntax/BumpPtrAllocator.swift

```diff
@@ -0,0 +1,147 @@
+//===---------- BumpPtrAllocator.swift - Bump-pointer Allocaiton ----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Bump-pointer allocation.
+@_spi(Testing)
+public class BumpPtrAllocator {
+  typealias Slab = UnsafeMutableRawBufferPointer
+
+  static private var SLAB_SIZE: Int = 4096
+  static private var GLOWTH_DELAY: Int = 128
+  static private var SLAB_ALIGNMENT: Int = 8
+
+  private var slabs: [Slab]
+  /// Points to the next unused address in `slabs.last`.
+  private var currentPtr: UnsafeMutableRawPointer?
+  /// Points to the end address of `slabs.last`. If `slabs` is not empty, this
+  /// is equivalent to `slabs.last!.baseAddress! + slabs.last!.count`
+  private var endPtr: UnsafeMutableRawPointer?
+  private var customSizeSlabs: [Slab]
+  private var _totalBytesAllocated: Int
```

Comment:
Why can’t this be `private(set) var totalByteSizeAllocated: Int` so we could remove the computed property below?

Reply:
Because I want `totalMemorySize` / `totalByteSizeAllocated` at the bottom of the file (because it's not a main functionality), and side by size, but I don't want to write a stored properties after the initializer.

Reply:
OK, sounds good 👍